File: c++/pyramid.cpp
/*
 * print like the following
 *     *
 *     * *
 *     * * *
 *     * * * *
 *     * * * * *
 */
#include <iostream>
using namespace std;

void simple_pypart(int n)
{
    // outer loop to handle number of rows
    //  n in this case
    for (int i=0; i<n; i++)
    {
        //  inner loop to handle number of columns
        //  values changing acc. to outer loop
        for(int j=0; j<=i; j++ )
        {
            // printing stars
            cout << "* ";
        }
 
        // ending line after each row
        cout << endl;
    }
}

/*
 * print like following:
 *              *
 *            * *
 *          * * *
 *        * * * *
 *      * * * * *
 */
void simple_pypart_reverse180(int n)
{
    // number of spaces
    int k = 2*n - 2;
 
    // outer loop to handle number of rows
    //  n in this case
    for (int i=0; i<n; i++)
    {
 
        // inner loop to handle number spaces
        // values changing acc. to requirement
        for (int j=0; j<k; j++)
            cout <<" ";
 
        // decrementing k after each loop
        k = k - 2;
 
        //  inner loop to handle number of columns
        //  values changing acc. to outer loop
        for (int j=0; j<=i; j++ )
        {
            // printing stars
            cout << "* ";
        }
 
        // ending line after each row
        cout << endl;
    }
}
/*
 * print like following:
 *          *
 *         * *
 *        * * *
 *       * * * *
 *      * * * * *
 */
void triangle(int n)
{
    // number of spaces
    int k = n - 1;
 
    // outer loop to handle number of rows
    //  n in this case
    for (int i=0; i<n; i++)
    {
        // inner loop to handle number spaces
        // values changing acc. to requirement
        for (int j=0; j<k; j++)
            cout <<" ";
 
        // decrementing k after each loop
        k = k - 1;
 
        //  inner loop to handle number of columns
        //  values changing acc. to outer loop
        for (int j=0; j<=i; j++ )
        {
            // printing stars
            cout << "* ";
        }
 
        // ending line after each row
        cout << endl;
    }
}
 
/*
 * print like following:
 *     1
 *     2 2
 *     3 3 3
 *     4 4 4 4
 *     5 5 5 5 5
 */
void numpat(int n)
{
    // initializing starting number
    int num = 1;
 
    // outer loop to handle number of rows
    //  n in this case
    for (int i=0; i<n; i++)
    {
        // re assigning num
        num = 1;
 
        //  inner loop to handle number of columns
        //  values changing acc. to outer loop
        for (int j=0; j<=i; j++ )
        {
            // printing number
            cout << num << " ";
 
            // incrementing number at each column
            num = num + 1;
        }
 
        // ending line after each row
        cout << endl;
    }
}

/*
 * print like following:
 *     1
 *     2  3
 *     4  5  6
 *     7  8  9  10
 *     11 12 13 14 15
 */
void numpat_without_reassigning(int n)
{
    // initialising starting number
    int num = 1;
 
    // outer loop to handle number of rows
    //  n in this case
    for (int i=0; i<n; i++)
    {
        // without re assigning num
        // num = 1;
 
        //  inner loop to handle number of columns
        //  values changing acc. to outer loop
        for (int j=0; j<=i; j++ )
        {
            // printing number
            cout << num << " ";
 
            // incrementing number at each column
            num = num + 1;
        }
 
        // ending line after each row
        cout << endl;
    }
}

/*
 * print like following:
 *     a
 *     b b
 *     c c c
 *     d d d d
 *     e e e e e
 */
void alphapat(int n)
{
    // initializing value corresponding to 'A'
    // ASCII value
    int num = 97;
 
    // outer loop to handle number of rows
    //  n in this case
    for (int i=0; i<n; i++)
    {
        //  inner loop to handle number of columns
        //  values changing acc. to outer loop
        for (int j=0; j<=i; j++ )
        {
            // explicitly converting to char
            char ch = char(num);
 
            // printing char value
            cout << ch << " ";
        }
 
        // incrementing number
        num = num + 1;
 
        // ending line after each row
        cout << endl;
    }
}

/*
 * print like following:
 *     A
 *     B C
 *     D E F
 *     G H I J
 *     K L M N O
 */
void contalpha(int n)
{
    // initializing value corresponding to 'A'
    // ASCII value
    int num = 65;
 
    // outer loop to handle number of rows
    //  n in this case
    for (int i=0; i<n; i++)
    {
        //  inner loop to handle number of columns
        //  values changing acc. to outer loop
        for (int j=0; j<=i; j++ )
        {
            // explicitely converting to char
            char ch = char(num);
 
            // printing char value
            cout << ch << " ";
 
            // incrementing number at each column
            num = num + 1;
        }
 
        // ending line after each row
        cout << endl;
    }
}

// Driver Function
int main()
{
    int n = 6;
    simple_pypart(n);
    simple_pypart_reverse180(n);
    triangle(n);
    numpat(n);
    numpat_without_reassigning(n);
    alphapat(n);
    contalpha(n);
    return 0;
}

File: lang/pylang/pyramid.py
#! /usr/bin/python3
# Function to demonstrate printing pattern
def pypart(n):
     
    # outer loop to handle number of rows
    # n in this case
    for i in range(0, n):
     
        # inner loop to handle number of columns
        # values changing acc. to outer loop
        for j in range(0, i+1):
            # printing stars
            print("* ",end="")
      
        # ending line after each row
        print("\r")
 
# Driver Code
n = 5
pypart(n)

# Function to demonstrate printing pattern
def pypart2(n):
     
    # number of spaces
    k = 2*n - 2
 
    # outer loop to handle number of rows
    for i in range(0, n):
     
        # inner loop to handle number spaces
        # values changing acc. to requirement
        for j in range(0, k):
            print(end=" ")
     
        # decrementing k after each loop
        k = k - 2
     
        # inner loop to handle number of columns
        # values changing acc. to outer loop
        for j in range(0, i+1):
         
            # printing stars
            print("* ", end="")
     
        # ending line after each row
        print("\r")

n = 5
pypart2(n)


#n to demonstrate printing pattern triangle
def triangle(n):
     
    # number of spaces
    k = 2*n - 2
 
    # outer loop to handle number of rows
    for i in range(0, n):
     
        # inner loop to handle number spaces
        # values changing acc. to requirement
        for j in range(0, k):
            print(end=" ")
     
        # decrementing k after each loop
        k = k - 1
     
        # inner loop to handle number of columns
        # values changing acc. to outer loop
        for j in range(0, i+1):
         
            # printing stars
            print("* ", end="")
     
        # ending line after each row
        print("\r")
 
# Driver Code
n = 5
triangle(n)

# Function to demonstrate printing pattern of numbers
def numpat(n):
     
    # initialising starting number 
    num = 1
 
    # outer loop to handle number of rows
    for i in range(0, n):
     
        # re assigning num
        num = 1
     
        # inner loop to handle number of columns
            # values changing acc. to outer loop
        for j in range(0, i+1):
         
                # printing number
            print(num, end=" ")
         
            # incrementing number at each column
            num = num + 1
     
        # ending line after each row
        print("\r")
 
# Driver code
n = 5
numpat(n)

# Function to demonstrate printing pattern of numbers
def contnum(n):
     
    # initializing starting number 
    num = 1
 
    # outer loop to handle number of rows
    for i in range(0, n):
     
        # not re assigning num
        # num = 1
     
        # inner loop to handle number of columns
        # values changing acc. to outer loop
        for j in range(0, i+1):
         
            # printing number
            print(num, end=" ")
         
            # incrementing number at each column
            num = num + 1
     
        # ending line after each row
        print("\r")
 
n = 5
 
# sending 5 as argument
# calling Function
contnum(n)

# Function to demonstrate printing pattern of alphabets
def alphapat(n):
     
    # initializing value corresponding to 'A' 
    # ASCII value
    num = 65
 
    # outer loop to handle number of rows
    # 5 in this case
    for i in range(0, n):
     
        # inner loop to handle number of columns
        # values changing acc. to outer loop
        for j in range(0, i+1):
         
            # explicitely converting to char
            ch = chr(num)
         
            # printing char value 
            print(ch, end=" ")
     
        # incrementing number
        num = num + 1
     
        # ending line after each row
        print("\r")
 
# Driver Code
n = 5
alphapat(n)

# Function to demonstrate printing pattern of alphabets
def  contalpha(n):
     
    # initializing value corresponding to 'A' 
    # ASCII value
    num = 65
 
    # outer loop to handle number of rows
    for i in range(0, n):
     
        # inner loop to handle number of columns
        # values changing acc. to outer loop
        for j in range(0, i+1):
         
            # explicitely converting to char
            ch = chr(num)
         
            # printing char value 
            print(ch, end=" ")
         
            # incrementing at each column
            num = num +1
     
     
        # ending line after each row
        print("\r")
 
# Driver code
n = 5
contalpha(n)
